src/Psl/DateTime/DateTime.php

```diff
@@ -0,0 +1,417 @@
+                $parts['seconds'],
+                $ns,
+            );
+        });
+    }
+
+    /**
+     * Parses a date-time string and returns a {@see DateTime} instance for the specified or default timezone.
+     *
+     * This method interprets a date-time string, potentially relative (e.g., "next Thursday", "10 minutes ago") or absolute
+     * (e.g., "2023-03-15 12:00:00"), into a {@see DateTime} object. An optional base time (`$relative_to`) can be provided to resolve
+     * relative date-time expressions. If `$relative_to` is not provided, the current system time is used as the base.
+     *
+     * @param string $raw_string The date-time string to parse.
+     * @param Timezone|null $timezone The timezone to use for the resulting DateTime instance. Defaults to the system's default timezone if null.
+     * @param TemporalInterface|null $relative_to The temporal context used to interpret relative date-time expressions. Defaults to the current system time if null.
+     *
+     * @throws Exception\InvalidArgumentException If parsing fails or the date-time string is invalid.
+     *
+     * @see https://www.php.net/manual/en/datetime.formats.php For information on supported date and time formats.
+     */
+    public static function parse(string $raw_string, ?Timezone $timezone = null, ?TemporalInterface $relative_to = null): self
+    {
+        $timezone ??= Timezone::default();
+
+        return self::fromTimestamp(
+            Timestamp::parse($raw_string, $timezone, $relative_to),
+            $timezone,
+        );
+    }
+
+    /**
+     * Returns the timestamp representation of this date time object.
+     *
+     * @mutation-free
+     */
+    public function getTimestamp(): Timestamp
+    {
+        return $this->timestamp;
+    }
+
+    /**
+     * Retrieves the year as an integer, following ISO-8601 conventions for numbering.
+     *
+     * This method returns the year part of the date. For years in the Anno Domini (AD) era, the returned value matches
+     * the Gregorian calendar year directly (e.g., 1 for AD 1, 2021 for AD 2021, etc.). For years before AD 1, the method
+     * adheres to the ISO-8601 standard, which does not use a year zero: 1 BC is represented as 0, 2 BC as -1, 3 BC as -2,
+     * and so forth. This ISO-8601 numbering facilitates straightforward mathematical operations on years across the AD/BC
+     * divide but may require conversion for user-friendly display or when interfacing with systems that use the traditional
+     * AD/BC notation.
+     *
+     * @return int The year, formatted according to ISO-8601 standards, where 1 AD is 1, 1 BC is 0, 2 BC is -1, etc.
+     *
+     * @mutation-free
+     */
+    public function getYear(): int
+    {
+        return $this->year;
+    }
+
+    /**
+     * Returns the month.
+     *
+     * @return int<1, 12>
+     *
+     * @mutation-free
+     */
+    public function getMonth(): int
+    {
+        return $this->month;
+    }
+
```

Comment:
maybe something like `getWeekOfYear()` could be handy here.

Reply:
we have `getIsoWeekNumber`, it returns [year, week_number], as according to the ISO standard, a week could belong to the previosu year, or next year, as explained in the method docblock.